kafka-commons/src/main/java/uk/gov/homeoffice/digital/sas/kafka/validators/SchemaValidator.java

```diff
@@ -0,0 +1,80 @@
+package uk.gov.homeoffice.digital.sas.kafka.validators;
+
+import static uk.gov.homeoffice.digital.sas.kafka.constants.Constants.KAFKA_SCHEMA_INCORRECT_FORMAT;
+import static uk.gov.homeoffice.digital.sas.kafka.constants.Constants.KAFKA_SCHEMA_INVALID;
+import static uk.gov.homeoffice.digital.sas.kafka.constants.Constants.KAFKA_SCHEMA_INVALID_RESOURCE;
+import static uk.gov.homeoffice.digital.sas.kafka.constants.Constants.KAFKA_SCHEMA_INVALID_VERSION;
+import static uk.gov.homeoffice.digital.sas.kafka.constants.Constants.KAFKA_SCHEMA_VALIDATED;
+import static uk.gov.homeoffice.digital.sas.kafka.constants.Constants.SCHEMA;
+
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+
+
+@Slf4j
+@Getter
+@Setter
+@NoArgsConstructor
+public abstract class SchemaValidator {
+
+  String resourceName;
+
+  List<String> validVersions;
+
+  public boolean isSchemaValid(String message) {
+    JsonObject jsonMessage = JsonParser.parseString(message).getAsJsonObject();
+    String schema = jsonMessage.get(SCHEMA).getAsString();
+    List<String> splitSchema = splitMessageSchema(schema);
+    return isValidMessage(splitSchema);
+  }
+
+  private List<String> splitMessageSchema(String schema) {
+    List<String> stringList = new ArrayList<>();
+    if (schema.contains(",")) {
+      stringList = Pattern.compile(", ")
+          .splitAsStream(schema).toList();
+    } else {
+      log.error(String.format(KAFKA_SCHEMA_INCORRECT_FORMAT, schema));
+    }
+    return stringList;
+  }
+
+  private boolean isValidMessage(List<String> splitSchema) {
+    if (splitSchema.size() == 2
+        && isValidResource(splitSchema.get(0))
```

Comment:
Is an unexpected resource the same as an unexpected version? Leads me to question what is versioned. I think its the entire schema that is versioned not just a version per resource schema. For instance if the consumer says it supports `^2.3.4` and the producer adds a new resource it might say it's a non breaking change and only increment the version to `2.4.0`. This would mean that the consumer would need to discard resources it didn't expect.

Reply:
The thought process behind checking the 'resource', was to make sure what the consumer was the right entity - I.E Time Entry. Currently the full schema info being sent across looks like -
"uk.gov.homeoffice.digital.sas.model.TimeEntry, 1.0.0"
We believed checking this would be beneficial reducing deserialisation errors. I know the ACL's should ensure the permissions and stop incorrect entities being sent. But adding this was a secondary check. 

Happy to discuss or remove this if we deem it unnecessary.

Reply:
As discussed Resource validation has been pulled out of kafka-commons and will be owned by the service.